archive-tar.c

```diff
@@ -38,9 +38,9 @@ static int write_tar_filter_archive(const struct archiver *ar,
 #endif
 
 /* writes out the whole block */
-static void write_block(const char * block) {
+static void write_block(const char *block) {
 	if (gzip) {
-		if (gzwrite(gzip, block, BLOCKSIZE) != BLOCKSIZE) error("gzwrite failed");
+		if (gzwrite(gzip, block, (unsigned) BLOCKSIZE) != BLOCKSIZE) error("gzwrite failed");
 	} else write_or_die(1, block, BLOCKSIZE);
```

Comment:
Please squash these two fixups into the previous commit, which introduces the bugs fixed here.